beeline/src/java/org/apache/hive/beeline/HiveSchemaTool.java

```diff
@@ -878,6 +886,189 @@ boolean validateColumnNullValues(Connection conn) throws HiveMetaException {
     }
   }
 
+  @VisibleForTesting
+  void createCatalog(String catName, String location, String description, boolean ifNotExists)
+      throws HiveMetaException {
+    catName = normalizeIdentifier(catName);
+    System.out.println("Create catalog " + catName + " at location " + location);
+    try (Connection conn = getConnectionToMetastore(false)) {
+      try (Statement stmt = conn.createStatement()) {
+        // If they set ifNotExists check for existence first, and bail if it exists.  This is
+        // more reliable then attempting to parse the error message from the SQLException.
+        if (ifNotExists) {
+          String query = "select " + quoteIf("NAME") + " from " + quoteIf("CTLGS") +
+              " where " + quoteIf("NAME") + " = '" + catName + "'";
+          ResultSet rs = stmt.executeQuery(query);
+          if (rs.next()) {
+            System.out.println("Catalog " + catName + " already exists");
+            return;
+          }
+        }
+        SQLGenerator sqlGenerator = new SQLGenerator(
+            DatabaseProduct.determineDatabaseProduct(
+                conn.getMetaData().getDatabaseProductName()
+            ), hiveConf);
+        String query = sqlGenerator.addForUpdateClause("select max(" + quoteIf("CTLG_ID") + ") " +
+            "from " + quoteIf("CTLGS"));
+        ResultSet rs = stmt.executeQuery(query);
```

Comment:
Don't we need to call - conn.setAutoCommit(false), so that "for update" holds the lock until commit ?

Reply:
Yup, probably.  I'll add this and create a new patch.